handlers.go

```diff
@@ -445,8 +304,13 @@ func (b *BouncerHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		reqParams.Lang = DefaultLang
 	}
 
-	isWinXpClient := isWindowsXPUserAgent(req.UserAgent())
-	isPre2024Stub := isPre2024StubUserAgent(req.UserAgent())
+	// If attribution_code is set, redirect to the stub service.
+	if b.shouldAttribute(reqParams) {
+		stubURL := b.stubAttributionURL(reqParams)
+		http.Redirect(w, req, stubURL, 302)
+		return
+	}
```

Comment:
Shouldn't this block be towards the end, after we've made changes to reqParams, so that the stub service URL is the right one for that client?  (I think that's an existing bug, so it should arguably get changed in its own PR)

Reply:
I might be wrong but attribution can theoretically only be set by mozorg, and (1) mozorg should never send the wrong params and (2) we largely avoid changing the params when the requests come from mozorg (kinda tied to (1))

Reply:
Fair.  I keep getting confused by this stuff.